### PR TITLE
Initial check in for vpn template

### PR DESCRIPTION
**Pull request by sourced-vince**

### CODE REVIEW

File: community/faas/templates/vpn.py.schema [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    description: |
+      Shared secret used to set the secure session between the Cloud VPN
+      gateway and the peer VPN gateway.
+
+outputs:
+  properties:
+    - targetVpnGateway:
+        type: string
+        description: Name of the Target VPN gateway resource.
+
```
Comment (tiadobatima1) [inline community/faas/templates/vpn.py.schema]: remove these blank lines? Other templates don't seem to have this

File: community/faas/tests/fixtures/configs/vpn.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  - path: templates/vpn.py
+    name: vpn.py
+
+resources:
+  - name: test-vpn-${RAND}
+    type: vpn.py
+    properties:
+      region: us-east1
+      network: network-${RAND}
+      peerAddress: 1.2.3.4
```
Comment (tiadobatima1) [inline community/faas/tests/fixtures/configs/vpn.yaml]: Are the tests passing with this IP?

File: community/faas/tests/integration/vpn.bats [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+function setup() {
+    # Global setup - this gets executed only once per test file
+    if [ ${BATS_TEST_NUMBER} -eq 1 ]; then
+        gcloud compute networks create "network-${RAND}" \
+            --project "${FAAS_PROJECT_ID}" \
+            --description "integration test ${RAND}" \
+            --subnet-mode custom
+        gcloud compute networks subnets create "subnet-${RAND}" \
```
Comment (tiadobatima1) [inline community/faas/tests/integration/vpn.bats]: don't need to specify the project here?

File: community/faas/tests/integration/vpn.bats [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            --network "network-${RAND}" \
+            --range 10.118.8.0/22 \
+            --region us-east1
+        gcloud compute routers create "router-${RAND}" \
```
Comment (tiadobatima1) [inline community/faas/tests/integration/vpn.bats]: Same comment about project. I can't quite understand how this is working :)

File: community/faas/tests/integration/vpn.bats [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        create_config
+    fi
+
+    # Per-test setup steps here
+}
+
+function teardown() {
+    # Global teardown - this gets executed only once per test file
+    if [[ "$BATS_TEST_NUMBER" -eq "${#BATS_TEST_NAMES[@]}" ]]; then
+        gcloud compute routers delete "router-${RAND}" --region us-east1 -q
```
Comment (tiadobatima1) [inline community/faas/tests/integration/vpn.bats]: missing project?

File: community/faas/tests/fixtures/configs/vpn.yaml (hunk 2 above)
Comment (sourced-vince) [reply community/faas/tests/fixtures/configs/vpn.yaml]: Yep. The tests do pass. I don't think they do any validation on actual IP's when creating the resource.

File: community/faas/templates/vpn.py.schema (hunk 1 above)
Comment (sourced-vince) [reply community/faas/templates/vpn.py.schema]: fixed.

File: community/faas/tests/integration/vpn.bats (hunk 3 above)
Comment (sourced-vince) [reply community/faas/tests/integration/vpn.bats]: I believe if you don't specify project in the commandline it uses the project that is part of your gcloud (gcloud config list). I can put it in for consistency here.

File: community/faas/tests/integration/vpn.bats (hunk 5 above)
Comment (sourced-vince) [reply community/faas/tests/integration/vpn.bats]: fixed

**Comment by sourced-vince** [comment]
fixed

File: community/faas/tests/integration/vpn.bats (hunk 3 above)
Comment (tiadobatima1) [reply community/faas/tests/integration/vpn.bats]: yep exactly. I think the main reason for being specific on this is that cloud operators will likely have their `gcloud` configured with the "project creator" project by default, not the test or service projects.

File: community/faas/tests/integration/vpn.bats [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        delete_config
+        rm -f "${RANDOM_FILE}"
+    fi
+
+    # Per-test teardown steps here
+}
+
+
+@test "Creating deployment ${DEPLOYMENT_NAME} from ${CONFIG}" {
+  gcloud deployment-manager deployments create "${DEPLOYMENT_NAME}" --config "${CONFIG}"
```
Comment (tiadobatima1) [inline community/faas/tests/integration/vpn.bats]: The gcloud commands from here to the end are also missing `--project`

**Pull request closed without merging**